kernel/src/process.rs

```diff
@@ -319,7 +366,10 @@ pub enum FaultResponse {
     /// Attempt to cleanup and restart the app which caused the fault. This
     /// resets the app's memory to how it was when the app was started and
     /// schedules the app to run again from its init function.
-    Restart,
+    ///
+    /// The provided restart policy is used to determine whether to reset the
+    /// app, and can be specified on a per-app basis.
+    Restart(&'static dyn ProcessRestartPolicy),
 
     /// Stop the app by no longer scheduling it to run.
     Stop,
```

Comment:
It seems kind of cleaner to have `Stop` be implemented as a `Restart` variant where the `ProcessRestartPolicy` always returns false, so that there is no need to verify that the `Restart(false)`branch of the match statement remains equivalent to the `Stop` branch.

But I can understand that `let FAULT_RESPONSE = FaultResponse::Restart(NeverRestart)` is maybe a little confusing compared to `let FAULT_RESPONSE = FaultResponse::Stop`, so I suppose I am fine with either.

Reply:
I tend to like the explicit "stop" option for clarity.